packages/astro/src/core/build/pipeline.ts

```diff
@@ -141,42 +140,23 @@ export class BuildPipeline extends Pipeline {
 	 * It collects the routes to generate during the build.
 	 * It returns a map of page information and their relative entry point as a string.
 	 */
-	retrieveRoutesToGenerate(): Map<PageBuildData, string> {
-		const pages = new Map<PageBuildData, string>();
+	retrieveRoutesToGenerate(): Map<RouteData, string> {
+		const pages = new Map<RouteData, string>();
 
-		for (const pageData of this.internals.pagesByKeys.values()) {
-			if (routeIsRedirect(pageData.route)) {
-				pages.set(pageData, pageData.component);
-			} else if (
-				routeIsFallback(pageData.route) &&
-				(i18nHasFallback(this.config) ||
-					(routeIsFallback(pageData.route) && pageData.route.route === '/'))
-			) {
```

Comment:
Here we removed essential logic that hasn't been placed anywhere else. Why? Maybe we should update the docs of this method to better align it with the new logic